Xamarin.PropertyEditing.Mac/Controls/CollectionInlineEditorControl.cs

```diff
@@ -25,8 +25,19 @@ public CollectionInlineEditorControl (IHostResourceProvider hostResources)
 			};
 
 			this.openCollection.Activated += (o, e) => {
-				CollectionEditorWindow.EditCollection (EffectiveAppearance, HostResources, ViewModel);
-				Window.MakeFirstResponder (this.openCollection);
+				NSWindow parentWindow = NSApplication.SharedApplication.KeyWindow;
+
+				var w = new CollectionEditorWindow (hostResources, ViewModel) {
+					Appearance = EffectiveAppearance
+				};
+
+				CocoaHelpers.RunModalForWindow (w, parentWindow, responseHandler: result => {
+					if (result != NSModalResponse.OK)
+						ViewModel.CancelCommand.Execute (null);
+					else
+						ViewModel.CommitCommand.Execute (null);
+					parentWindow?.MakeFirstResponder (this.openCollection);
```

Comment:
Remove this line here, and instead pass in the `NSView` that we want to refocus. I'll add the suggestion there. This might make it a bit more usable too.